src/platform/input/game-pads.js

```diff
@@ -134,111 +935,142 @@ class GamePads {
-        return this.current[index].pad.buttons[pc[key]].pressed;
+    isPressed(orderIndex, button) {
+        return this.current[orderIndex]?.isPressed(button) || false;
     }
 
     /**
      * Returns true if the button was pressed since the last frame.
      *
-     * @param {number} index - The index of the pad to check, use constants {@link PAD_1},
-     * {@link PAD_2}, etc.
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
      * @param {number} button - The button to test, use constants {@link PAD_FACE_1}, etc.
      * @returns {boolean} True if the button was pressed since the last frame.
      */
-    wasPressed(index, button) {
-        if (!this.current[index]) {
-            return false;
-        }
-
-        const key = this.current[index].map.buttons[button];
-        const i = pc[key];
-
-        // Previous pad buttons may not have been populated yet
-        // If this is the first time frame a pad has been detected
-        return this.current[index].pad.buttons[i].pressed && !(this.previous[index] && this.previous[index][i]);
+    wasPressed(orderIndex, button) {
+        return this.current[orderIndex]?.wasPressed(button) || false;
     }
 
     /**
      * Returns true if the button was released since the last frame.
      *
-     * @param {number} index - The index of the pad to check, use constants {@link PAD_1},
-     * {@link PAD_2}, etc.
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
      * @param {number} button - The button to test, use constants {@link PAD_FACE_1}, etc.
      * @returns {boolean} True if the button was released since the last frame.
      */
-    wasReleased(index, button) {
-        if (!this.current[index]) {
-            return false;
-        }
+    wasReleased(orderIndex, button) {
+        return this.current[orderIndex]?.wasReleased(button) || false;
+    }
 
-        const key = this.current[index].map.buttons[button];
-        const i = pc[key];
+    /**
+     * Get the value of one of the analog axes of the pad.
+     *
+     * @param {number} orderIndex - The index of the pad to check, use constants {@link PAD_1}, {@link PAD_2}, etc. For gamepad index call the function from the pad.
+     * @param {number} axis - The axis to get the value of, use constants {@link PAD_L_STICK_X}, etc.
+     * @returns {number} The value of the axis between -1 and 1.
+     */
+    getAxis(orderIndex, axis) {
+        return this.current[orderIndex]?.getAxis(axis) || 0;
+    }
 
-        // Previous pad buttons may not have been populated yet
-        // If this is the first time frame a pad has been detected
-        return !this.current[index].pad.buttons[i].pressed && (this.previous[index] && this.previous[index][i]);
+    /**
+     * Make the gamepad vibrate.
```

Comment:
As far as I'm aware, vibration is not widely supported in browsers (as of today). Worth putting in a note about what happens when the underlying browser doesn't support the operation?

Reply:
https://caniuse.com/mdn-api_gamepadhapticactuator_pulse

Reply:
In fact only Safari doesn't support it. And if it could not complete it will result in `Promise<false>` so I'm not sure that's really needed.

Reply:
Oh really? But the 'caniuse' page seems to indicate it's Firefox only, no?

Reply:
Because you are looking at the `pulse` function which is only supported by Firefox. Other browsers are using `playEffect`.

Reply:
And as `pulse` name is more explicit than `playEffect` I used `pulse` as name for the function. But in the code if you look at it, it selects the proper API function.

Reply:
Oh, ok, got it.